File: UPTMDigital.API/Models/Profesor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UPTMDigital.API.Models
{
    [Table("Profesor")]
    public class Profesor
    {
        [Key]
        public int IdProfesor { get; set; }
        public string Cedula { get; set; } = null!;
        public string Nombres { get; set; } = null!;
        public string Apellidos { get; set; } = null!;
        public string? CorreoInstitucional { get; set; }
        public string? CodProfesor { get; set; }
        public string? Departamento { get; set; }
        public string? Telefono { get; set; }

        // --- FK Normalizada ---
        public int? UsuarioId { get; set; }
        [ForeignKey("UsuarioId")]
        public Usuario? Usuario { get; set; }

        public bool Activo { get; set; } = true;
    }
}

File: uptmdigital_app/lib/models/profesor.dart
class Profesor {
  final int idProfesor;
  final String cedula;
  final String nombres;
  final String apellidos;
  final String correoInstitucional;
  final String departamento;
  final int? usuarioId;
  final String? profileImageUrl;
  final String? telefono;

  Profesor({
    required this.idProfesor,
    required this.cedula,
    required this.nombres,
    required this.apellidos,
    required this.correoInstitucional,
    required this.departamento,
    this.usuarioId,
    this.profileImageUrl,
    this.telefono,
  });

  factory Profesor.fromJson(Map<String, dynamic> json) {
    return Profesor(
      idProfesor: json['idProfesor'] ?? 0,
      cedula: json['cedula'] ?? '',
      nombres: json['nombres'] ?? '',
      apellidos: json['apellidos'] ?? '',
      correoInstitucional: json['correoInstitucional'] ?? '',
      departamento: json['departamento'] ?? '',
      usuarioId: json['usuarioId'],
      profileImageUrl: json['profileImageUrl'],
      telefono: json['telefono'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idProfesor': idProfesor,
      'cedula': cedula,
      'nombres': nombres,
      'apellidos': apellidos,
      'correoInstitucional': correoInstitucional,
      'departamento': departamento,
      'usuarioId': usuarioId,
      'profileImageUrl': profileImageUrl,
      'telefono': telefono,
    };
  }
}
